### PR TITLE
Add delay between writing file and modifying it

### PR DESCRIPTION
**Pull request by lukechurch**
@johnmccutchan
Addresses: https://github.com/flutter/flutter/issues/6964

Timing issue on OS X devfs between creating and modifying a file

### CODE REVIEW

**Comment by johnmccutchan** [comment]
LGTM

File: packages/flutter_tools/test/devfs_test.dart [hunk 1]
```diff
@@ -39,18 +40,21 @@ void main() {
...
-    });
     testUsingContext('add new file to local file system', () async {
       File file = new File(path.join(basePath, filePath2));
       await file.parent.create(recursive: true);
       file.writeAsBytesSync(<int>[1, 2, 3, 4, 5, 6, 7]);
       await devFS.update();
       expect(devFSOperations.contains('writeFile test foo/bar.txt'), isTrue);
+      // Need to delay between when we first write to a file and when
+      // we modify it.
+      await new Future<Null>.delayed(new Duration(seconds: 1));
```
Comment (Hixie) [inline packages/flutter_tools/test/devfs_test.dart]: Please revert this. We should not put sleep statements in our tests. That slows down the tests.

What is the underlying problem here?

Comment (johnmccutchan) [inline packages/flutter_tools/test/devfs_test.dart]: The underyling problem is that file modification timestamps on MacOS are not high resolution enough. The flow in the test is:

1) Write a new file.
2) Modify the file.
3) Verify that we detect the modification.

Putting a sleep between (1) and (2) makes the mtimes different.

Comment (johnmccutchan) [inline packages/flutter_tools/test/devfs_test.dart]: @Hixie I don't think that is possible via dart:io

Comment (johnmccutchan) [inline packages/flutter_tools/test/devfs_test.dart]: That's an alternative that may work. It may also be just as racey though. @lukechurch

Comment (Hixie) [inline packages/flutter_tools/test/devfs_test.dart]: I mean to set the time in the past. You can pass a date to `touch` to make it old, then modify it and see if the time changed (or whatever it is you're testing).

Comment (lukechurch) [inline packages/flutter_tools/test/devfs_test.dart]: Sorry, I was out yesterday. How do we make a scheme dependant on ```touch``` work on Windows?

Comment (abarth) [inline packages/flutter_tools/test/devfs_test.dart]: Working on Windows is not a requirement at the moment.

Comment (Hixie) [inline packages/flutter_tools/test/devfs_test.dart]: Hopefully by the time we support windows, dart:io will expose the relevant syscalls. (https://github.com/dart-lang/sdk/issues/27877)

**Pull request merged**